bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/WebThingHandler.java

```diff
@@ -0,0 +1,302 @@
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WebThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gregor Roth - Initial contribution
+ */
+@NonNullByDefault
+public class WebThingHandler extends BaseThingHandler implements ChannelHandler {
+    private static final Duration RECONNECT_PERIOD = Duration.ofHours(23);
+    private static final Duration HEALTH_CHECK_PERIOD = Duration.ofSeconds(70);
+    private static final ItemChangedListener EMPTY_ITEM_CHANGED_LISTENER = (channelUID, stateCommand) -> {
+    };
+
+    private final Logger logger = LoggerFactory.getLogger(WebThingHandler.class);
+    private final HttpClient httpClient;
+    private final WebSocketClient webSocketClient;
+    private final AtomicBoolean isActivated = new AtomicBoolean(true);
+    private final Map<ChannelUID, ItemChangedListener> itemChangedListenerMap = new ConcurrentHashMap<>();
+    private final AtomicReference<Optional<ConsumedThing>> webThingConnectionRef = new AtomicReference<>(
+            Optional.empty());
+    private final AtomicReference<Instant> lastReconnect = new AtomicReference<>(Instant.now());
+    private final AtomicReference<Optional<ScheduledFuture<?>>> watchdogHandle = new AtomicReference<>(
+            Optional.empty());
+    private @Nullable URI webThingURI = null;
+
+    public WebThingHandler(Thing thing, HttpClient httpClient, WebSocketClient webSocketClient) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.webSocketClient = webSocketClient;
+    }
+
+    private boolean isOnline() {
+        return getThing().getStatus() == ThingStatus.ONLINE;
+    }
+
+    private boolean isDisconnected() {
+        return (getThing().getStatus() == ThingStatus.OFFLINE) || (getThing().getStatus() == ThingStatus.UNKNOWN);
+    }
+
+    private boolean isAlive() {
+        return webThingConnectionRef.get().map(ConsumedThing::isAlive).orElse(false);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        isActivated.set(true); // set with true, even though the connect may fail. In this case retries will be
+                               // triggered
+
+        // perform connect in background
+        scheduler.execute(() -> {
+            // WebThing URI present?
+            this.webThingURI = readWebThingURI();
+            if (webThingURI != null) {
+                logger.debug("try to connect WebThing {}", webThingURI);
+                var connected = tryReconnect(webThingURI);
+                if (connected) {
+                    logger.debug("WebThing {} connected", getWebThingLabel());
+                }
+            } else {
+                logger.warn("could not initialize WebThing. URI is not set or invalid. {}",
+                        getConfigAs(WebThingConfiguration.class).webThingURI);
```

Comment:
Please use `this.webThingURI` here, as `getConfigAs()` is an expensive operation.

Reply:
done